src/ParamConverter/CreateCardParamConverter.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\ParamConverter;
+
+use Sensio\Bundle\FrameworkExtraBundle\Request\ParamConverter\ParamConverterInterface;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\ParamConverter;
+use Symfony\Component\HttpFoundation\Request;
+use App\Validator\CardDataValidator;
+use App\Repository\CardRepository;
+use App\Request\CreateCard;
+use Symfony\Component\HttpFoundation\Exception\BadRequestException;
+
+class CreateCardParamConverter implements ParamConverterInterface
+{
+    private $validator;
+
+    public function __construct(CardRepository $repository, CardDataValidator $validator)
+    {
+        $this->validator = $validator;
+        $this->cardRepository = $repository;
+    }
+
+    public function apply(Request $request, ParamConverter $configuration)
+    {
+        $content = json_decode($request->getContent(), true, 512, JSON_THROW_ON_ERROR);
+
+        if (
+            $this->validator->isCardDataValid(
```

Comment:
I suggest something like:
```
$cardDTO = new CreateCard(
     $this->getName($requestContent),
     ... // and so on
);
```
So for each of property create a method that extract value from jsonContent. And also i that methods perform validation. So you could make it more readable. Also after that you could get rid of `isCardDataValid()`
And thanks to that you could set BadRequest message for each error so user get more explicite information whats wrong.

Reply:
done please verify